docs/html/man/man3/Image.3
.TH "Image" 3 "8 Jul 2009" "Version 0.113" "q4wine" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Image \- This class provide database functions for \fBImage\fP table.  

.PP
.SH SYNOPSIS
.br
.PP
\fC#include <src/core/database/image.h>\fP
.PP
Inherits \fBPrefix\fP.
.PP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBImage\fP ()"
.br
.RI "\fIClass constructor. \fP"
.ti -1c
.RI "QList< QStringList > \fBgetFields\fP (void) const "
.br
.RI "\fIGetting images list. \fP"
.ti -1c
.RI "QString \fBgetPath\fP (const QString name) const "
.br
.RI "\fIGetting file path reccord by image name. \fP"
.ti -1c
.RI "bool \fBisExistsByName\fP (const QString name) const "
.br
.RI "\fIChek for image existance by name. \fP"
.ti -1c
.RI "bool \fBaddImage\fP (const QString name, const QString path) const "
.br
.RI "\fIAdd image reccord. \fP"
.ti -1c
.RI "bool \fBdelImage\fP (const QString name) const "
.br
.ti -1c
.RI "bool \fBrenameImage\fP (const QString name, const QString old_name) const "
.br
.in -1c
.SH "Detailed Description"
.PP 
This class provide database functions for \fBImage\fP table. 
.PP
Definition at line 50 of file image.h.
.SH "Member Function Documentation"
.PP 
.SS "bool Image::addImage (const QString name, const QString path) const"
.PP
Add image reccord. 
.PP
\fBParameters:\fP
.RS 4
\fIname\fP image name 
.br
\fIpath\fP image file path 
.RE
.PP
\fBReturns:\fP
.RS 4
If image exists -- returns true. 
.RE
.PP

.PP
Definition at line 86 of file image.cpp.
.SS "QList< QStringList > Image::getFields (void) const"
.PP
Getting images list. 
.PP
\fBReturns:\fP
.RS 4
Return QList on QStringList containing images. 
.RE
.PP

.PP
Reimplemented from \fBPrefix\fP.
.PP
Definition at line 37 of file image.cpp.
.SS "QString Image::getPath (const QString name) const"
.PP
Getting file path reccord by image name. 
.PP
\fBParameters:\fP
.RS 4
\fIname\fP image name 
.RE
.PP
\fBReturns:\fP
.RS 4
Return QList on QStringList containing images. 
.RE
.PP

.PP
Reimplemented from \fBPrefix\fP.
.PP
Definition at line 58 of file image.cpp.
.SS "bool Image::isExistsByName (const QString name) const"
.PP
Chek for image existance by name. 
.PP
\fBParameters:\fP
.RS 4
\fIname\fP image name 
.RE
.PP
\fBReturns:\fP
.RS 4
If image exists -- returns true. 
.RE
.PP

.PP
Reimplemented from \fBPrefix\fP.
.PP
Definition at line 71 of file image.cpp.

.SH "Author"
.PP 
Generated automatically by Doxygen for q4wine from the source code.


src/core/database/image.cpp
/***************************************************************************
 *   Copyright (C) 2008, 2009 by Malakhov Alexey                           *
 *   brezerk@gmail.com                                                     *
 *                                                                         *
 *   This program is free software: you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation, either version 3 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
 *                                                                         *
 *   In addition, as a special exception, the copyright holders give       *
 *   permission to link the code of this program with any edition of       *
 *   the Qt library by Trolltech AS, Norway (or with modified versions     *
 *   of Qt that use the same license as Qt), and distribute linked         *
 *   combinations including the two.  You must obey the GNU General        *
 *   Public License in all respects for all of the code used other than    *
 *   Qt.  If you modify this file, you may extend this exception to        *
 *   your version of the file, but you are not obligated to do so.  If     *
 *   you do not wish to do so, delete this exception statement from        *
 *   your version.                                                         *
 ***************************************************************************/

#include "image.h"

Image::Image()
{
  	this->_TABLE="image";
}

QList<QStringList> Image::getFields(void) const{
	QList<QStringList> valuelist;
	//                      0     1
	QSqlQuery query("SELECT name, path FROM images ORDER BY name");
	if (query.exec()){
		while (query.next()) {
			QStringList values;
			int i=0;
			while (query.value(i).isValid()){
				values.append(query.value(i).toString());
				i++;
			}
			valuelist.append(values);
		}
	} else {
		qDebug()<<"SqlError: "<<query.lastError();
	}
	return valuelist;
}


QString Image::getPath(const QString name) const{
	QString result;
	QSqlQuery query("SELECT path FROM images WHERE name=:name ORDER BY name");
	query.bindValue(":name", name);
	if (query.exec()){
		query.next();
		result = query.value(0).toString();
	} else {
		qDebug()<<"SqlError: "<<query.lastError();
	}
	return result;
}

bool Image::isExistsByName(const QString name) const{
	QSqlQuery query;
	query.prepare("SELECT id FROM images WHERE name=:name");
	query.bindValue(":name", name);
	if (!query.exec()){
		qDebug()<<"SqlError: "<<query.lastError()<<query.executedQuery();
		return false;
	}
	query.first();
	if (query.isValid()){
		return true;
	}
	return false;
}

bool Image::addImage(const QString name, const QString path) const{
  	QSqlQuery query;
	query.prepare("INSERT INTO images(name, path) VALUES(:name, :path)");
	query.bindValue("name", name);
	query.bindValue("path", path);

	if (!query.exec()){
		qDebug()<<"SqlError: "<<query.lastError()<<query.executedQuery();
		return false;
	}
	return true;
}

bool Image::renameImage(const QString name, const QString old_name) const{
  	QSqlQuery query;
	query.prepare("UPDATE images SET name=:name WHERE name=:old_name");
	query.bindValue("name", name);
	query.bindValue("old_name", old_name);

	if (!query.exec()){
		qDebug()<<"SqlError: "<<query.lastError()<<query.executedQuery();
		return false;
	}
	return true;
}

bool Image::delImage(const QString name) const{
  	QSqlQuery query;
	query.prepare("DELETE FROM images WHERE name=:name");
	query.bindValue("name", name);

	if (!query.exec()){
		qDebug()<<"SqlError: "<<query.lastError()<<query.executedQuery();
		return false;
	}
	return true;
}

